packages/components/tag/src/tag-list.scss

```diff
@@ -1,89 +1,76 @@
 :host {
+  --_border-color: var(--sl-color-tag-subtle-idle-border);
   --_gap: var(--sl-space-tag-list-md);
-  --_group-font: var(--sl-text-tag-label-md);
-  --_group-margin: calc(var(--_group-space) * 4);
-  --_group-margin-more: calc(var(--_group-space) * 2);
-  --_group-space: var(--sl-space-tag-counter-gap);
-  --_group-width: var(--sl-size-tag-counter);
-  --_stacked-height: var(--sl-size-tag-md);
 
   display: flex;
-  flex-shrink: 0;
-  gap: var(--sl-space-tag-list-md);
+  gap: var(--_gap);
 }
 
 :host([stacked]) {
   flex-wrap: nowrap;
 
   .list {
     align-content: flex-end;
-    display: flex;
     flex-wrap: nowrap;
-    gap: var(--_gap);
     overflow: hidden;
   }
 }
 
-:host([size='lg']) {
-  --_stacked-height: var(--sl-size-tag-lg);
+:host([emphasis='bold']) {
+  --_border-color: var(--sl-color-tag-bold-idle-border);
 }
 
-.group {
-  sl-tag {
-    font: var(--_group-font);
-    position: relative;
-
-    &:not([label='1']) {
-      margin-inline-start: calc(var(--_group-margin-more) * -1);
-    }
-
-    &:not([label='1'])::before {
-      inset-inline-start: var(--_group-margin-more);
-    }
-
-    &:not([label='1'], [label='2']) {
-      margin-inline-start: calc(var(--_group-margin) * -1);
-    }
+::slotted(*) {
+  flex-shrink: 0;
+}
 
-    &:not([label='1'], [label='2'])::after {
-      inset-inline-start: var(--_group-margin);
-    }
+.stack {
+  display: flex;
+  position: relative;
 
-    &:not([label='1'])::before,
-    &:not([label='1'], [label='2'])::after {
-      --_border-color: var(--sl-color-tag-subtle-idle-border);
+  &.double,
+  &.triple {
+    border-inline-start: 1px solid #fff;
+  }
 
-      block-size: 100%;
-      border-inline-start: calc(var(--_group-width) * 0.45) solid var(--_border-color);
-      border-radius: var(--sl-border-radius-default);
-      content: '';
-      inline-size: 100%;
-      position: absolute;
-    }
+  &.double {
+    padding-inline-start: 2px;
   }
 
-  sl-tag[emphasis='bold'] {
-    &:not([label='1'])::before,
-    &:not([label='1'], [label='2'])::after {
-      --_border-color: var(--sl-color-tag-bold-idle-border);
-    }
+  &.triple {
+    padding-inline-start: 4px;
   }
 
-  sl-tag:focus {
-    &:not([label='1'])::before,
-    &:not([label='1'], [label='2'])::after {
-      --_border-color: var(--sl-color-tag-subtle-active-border);
-    }
+  &:has(sl-tag:focus) {
+    --_border-color: var(--sl-color-tag-subtle-active-border);
   }
 
-  sl-tag[emphasis='bold']:focus {
-    &:not([label='1'])::before,
-    &:not([label='1'], [label='2'])::after {
-      --_border-color: var(--sl-color-tag-bold-active-border);
-    }
+  &:has(sl-tag[emphasis='bold']:focus) {
+    --_border-color: var(--sl-color-tag-subtle-active-border);
   }
 }
 
+.double::before,
+.triple::before,
+.triple::after {
+  border-color: var(--_border-color);
+  border-inline-start: 2px solid var(--_border-color);
+  border-radius: 4px;
```

Comment:
Can you please use tokens here? :)

Reply:
We could have problems with shape in other themes when we will leave the exact value here, not a token?